Add collection and formula-related samples (#158)
OfficeDev/office-scripts-docs: docs/resources/excel-samples.md

}
```

### Querying and deleting from a collection

This script creates a new worksheet. It checks for an existing copy of the worksheet and deletes it before making a new sheet.

```typescript
function main(workbook: ExcelScript.Workbook) {
  // Name of the worksheet to be added.
  let name = "Index";

  // Get any worksheet with that name.
  let sheet = workbook.getWorksheet("Index");
  

>   // If `null` wasn't returned, then there was a pre-existing worksheet with that name.

Reviewer: ```suggestion
  // If `null` wasn't returned, then there's already a worksheet with that name.
```

}
```

## Formulas

These samples use Excel formulas and show how to work with them in scripts.

## Single formula

This script sets a cell's formula, then displays how Excel stores the cell's formula and value separately.

```typescript
function main(workbook: ExcelScript.Workbook) {
  let selectedSheet = workbook.getActiveWorksheet();

  // Set A1 to 2.
  let a1 = selectedSheet.getRange("A1");
  a1.setValue(2);

  // Set B1 to the formula =(2*A1), which should equal 4.
  let b1 = selectedSheet.getRange("B1")
  b1.setFormula("=(2*A1)");

  // Log the current results for `getFormula` and `getValue` at B1.
  console.log(`B1 - Formula: ${b1.getFormula()} | Value: ${b1.getValue()}`);
}
```

### Spilling results from a formula

> This script transposes the range "A1:D2" to "A4:B7" by using TRANSPOSE function. If the transpose results in #SPILL error, it clears the target range and applies the formula again.

Reviewer: ```suggestion
This script transposes the range "A1:D2" to "A4:B7" by using the TRANSPOSE function. If the transpose results in a #SPILL error, it clears the target range and applies the formula again.
```

```typescript
function main(workbook: ExcelScript.Workbook) {
  let sheet = workbook.getActiveWorksheet();
  // Use the data in A1:D2 for the sample.
  let dataAddress = "A1:D2"
  let inputRange = sheet.getRange(dataAddress);

  // Place the transposed data starting at A4.
  let targetStartCell = sheet.getRange("A4");

  // Compute the target range.
  let targetRange = targetStartCell.getResizedRange(inputRange.getColumnCount() - 1, inputRange.getRowCount() - 1);

  // Call the transpose helper function.
  targetStartCell.setFormula(`=TRANSPOSE(${dataAddress})`);

  // Check if the range update resulted in a spill error.
  let checkValue = targetStartCell.getValue() as string;
  if (checkValue === '#SPILL!') {

>     // Clear target range and call the transpose function again.

Reviewer: ```suggestion
    // Clear the target range and call the transpose function again.
```
